packages/hardware-ledger/src/LedgerKeyAgent.ts

```diff
@@ -134,7 +145,13 @@ const isMultiSig = (tx: Transaction): boolean => {
 type LedgerConnectionWithCommunicationTypeAndDevicePath = {
   deviceConnection: LedgerConnection;
   communicationType: CommunicationType;
-  devicePath?: string;
+  nodeHidDevicePath?: string;
```

Comment:
Renamed `devicePath` to `nodeHidDevicePath` to indicate the device path exists only in the node js environment.

Reply:
A union type based on `communicationType` would be better

Reply:
In this case I would prefer to keep a slightly loose type. If we go with a union this will affect other places where we will have to check for the value of the communicationType to access particular properties which will complicate implementation. IMO it is pretty safe to keep this type as is.

Reply:
I don't see this as a complication, but as an assurance (by tsc) that code can handle different types of connection objects

Reply:
oki doki, let me attempt it and let's evaluate the result 😉

Reply:
@mkazlauskas Please take a look on those two approaches:
1. discriminated union which you suggested: https://github.com/input-output-hk/cardano-js-sdk/compare/feat/lw-9808-lace-HW-onboarding-revamp...feat/lw-9808-lace-HW-onboarding-revamp-a
2. simplification of the type: https://github.com/input-output-hk/cardano-js-sdk/compare/feat/lw-9808-lace-HW-onboarding-revamp...feat/lw-9808-lace-HW-onboarding-revamp-b

From the functionality perspective, they are equal - they give the same result reusing remembered connection
From the simplicity/maintainability perspective, I favor the second solution. IMO it is simpler for other so work with this implementation.

What are your thoughts?

Reply:
I like the union because it is safer, less combinations to worry about when using the object